src/chains/filecoin/filecoin/tests/things/address.test.ts

```diff
@@ -0,0 +1,145 @@
+        Type: "bls",
+        PrivateKey: "d9uNVQQ7Ek+Ri8X0gLBMKslCCN28PfLVhS8b6ZJZUhY="
+      },
+      testInfo: {
+        // this is just Buffer.from(lotusKeyInfo.PrivateKey, "base64").toString("hex")
+        privateKey:
+          "77db8d55043b124f918bc5f480b04c2ac94208ddbc3df2d5852f1be992595216",
+        publicAddress:
+          "t3sym5jz44yhcfslttqfsygrkanh5w556huzupuuz5ox44dniehguj6w23gbnxxp4ztgwxaesr2wtvdwkr4jyq"
+      }
+    };
+
+    const secpAddress = {
+      lotusKeyInfo: {
+        Type: "secp256k1",
+        PrivateKey: "hJ5suaM/nJwkclh/QoE59CFbWIGZEgPIhKapR68QqkE="
+      },
+      testInfo: {
+        // this is just Buffer.from(lotusKeyInfo.PrivateKey, "base64").toString("hex")
+        privateKey:
+          "849e6cb9a33f9c9c2472587f428139f4215b5881991203c884a6a947af10aa41",
+        publicAddress: "t15splylflm24lhvigtcsi4sjsagz6f7cwthnylki"
+      }
+    };
+
+    it("should derive a real BLS address from a private key", async () => {
+      const address = Address.fromPrivateKey(blsAddress.testInfo.privateKey);
+
+      assert.strictEqual(address.value, blsAddress.testInfo.publicAddress);
+
+      const address2 = Address.fromPrivateKey(
+        blsAddress.testInfo.privateKey,
+        AddressProtocol.BLS
+      );
+
+      assert.strictEqual(address2.value, address.value);
+    });
+
+    it("should derive a real SECP256K1 address from a private key", async () => {
+      const address = Address.fromPrivateKey(
+        secpAddress.testInfo.privateKey,
+        AddressProtocol.SECP256K1
+      );
+
+      assert.strictEqual(address.value, secpAddress.testInfo.publicAddress);
+    });
+
+    it("should create a random address when calling Address.random()", async () => {
+      const address = Address.random();
+
+      assert(Address.isValid(address.value));
+    });
+
+    it("properly signs a buffer with a BLS address", async () => {
+      const address = Address.fromPrivateKey(
+        blsAddress.testInfo.privateKey,
+        AddressProtocol.BLS
+      );
+      const buffer = Buffer.from([0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55]);
+      const signature = await address.signBuffer(buffer);
+      assert.strictEqual(
+        signature.toString("base64"),
+        "pmfBJoCCDKhNsp2CzneA5Q+gDmZX9q2r6t1MeXgyLDkFdQw5KgdeQzr+ZInZttpFAhw9wjsiLEqV+agIZM1wxuWpcbpb7sz73XiloKFj20BkP5yvyC/ub+MFIWREFlL2"
+      );
+    });
+
+    it("properly signs a buffer with a SECP256K1 address", async () => {
+      const address = Address.fromPrivateKey(
+        secpAddress.testInfo.privateKey,
+        AddressProtocol.SECP256K1
+      );
+      const buffer = Buffer.from([0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55]);
+      const signature = await address.signBuffer(buffer);
+      assert.strictEqual(
+        signature.toString("base64"),
+        "I9Fk7jXdAPIi8AGQgwZe+T+CAsESdPcIfiK7qWRrcLl/G4+Ol0jB2TM1IDglvxuhM6Xf9G7zGwVrN9glNyCNDAE="
+      );
+    });
+
+    // This test fails and I can't figure out why, but it's not worth the time now.
```

Comment:
mind adding should we add a `// TODO` to these? As well as in the `signMessage` comment? Or maybe open issues?

Reply:
Added comments and opened https://github.com/trufflesuite/ganache-core/issues/867